s3/src/test/scala/com/meltwater/fawn/s3/S3PayloadsSpec.scala

```diff
@@ -0,0 +1,281 @@
+package com.meltwater.fawn.s3
+
+import cats.effect.IO
+import net.andimiller.munit.cats.effect.styles.FlatIOSpec
+import org.http4s.scalaxml.xml
+import org.http4s.{EntityDecoder, Header, Response}
+
+import scala.xml.Elem
+
+class S3PayloadsSpec extends FlatIOSpec {
+
+  val listBucketsExample: String =
+    """<?xml version="1.0" encoding="UTF-8"?>
+      |<ListAllMyBucketsResult>
+      | <Buckets>
+      |  <Bucket>
+      |    <CreationDate>2022-01-17</CreationDate>
+      |    <Name>bucket-name</Name>
+      |  </Bucket>
+      | </Buckets>
+      | <Owner>
+      | <DisplayName>accountname</DisplayName>
+      | <ID>accountid</ID>
+      | </Owner>
+      |</ListAllMyBucketsResult>""".stripMargin
+
+  test("decode a list all my buckets response") {
+    EntityDecoder[IO, Elem]
+      .decode(
+        Response[IO]()
+          .withEntity(listBucketsExample)
+          .withHeaders(Header("content-type", "text/xml")),
+        false
+      )
+      .value
+      .flatMap { r =>
+        IO {
+          r.map { elem =>
+            ListBucketsResponse.xmlDecoder.read(elem)
+          }.isRight
+        }.assertEquals(true)
+      }
+  }
+
+  val listObjectsExample: String =
+    """<?xml version="1.0" encoding="UTF-8"?>
+      |<ListBucketResponse>
+      | <Name>bucket</Name>
+      | <KeyCount>205</KeyCount>
+      | <MaxKeys>1000</MaxKeys>
+      | <IsTruncated>false</IsTruncated>
+      | <Contents>
+      |  <Key>my-image.jpg</Key>
+      |  <LastModified>2009-10-12T17:50:30.000Z</LastModified>
+      |  <ETag>"fba9dede5f27731c9771645a39863328"</ETag>
+      |  <Size>434234</Size>
+      |  <StorageClass>STANDARD</StorageClass>
+      | </Contents>
+      |</ListBucketResponse>""".stripMargin
+
+  test("decode a list all objects within a bucket response") {
+    EntityDecoder[IO, Elem]
+      .decode(
+        Response[IO]()
+          .withEntity(listObjectsExample)
+          .withHeaders(Header("content-type", "text/xml")),
```

Comment:
prefer `.transformHeaders(_.put(Header(...)))` so it doesn't overwrite existing headers